### PR TITLE
fix(Core/Unit): Improve check for players to automatically stand if a…

### PR DESCRIPTION
**Pull request by Kitzunu**
…ttacked

## Changes Proposed:
-  title  

## Tests Performed:

- none

## How to Test the Changes:

1. Go in combat with a player that is not standing

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by tkn963** [comment]
I can't seem to test this because I don't have any issues using the latest commit of the master branch. I tried sitting down as alliance and attacking as horde and I tried in duels but every time the player being attacked stands up. Melee attacks and spells that hit and that misses causes the player to stand up for me.

**Comment by Kitzunu** [comment]
that's correct

**Comment by Kitzunu** [comment]
if attacked, a player should stand up

**Comment by tkn963** [comment]
> 
> 
> if attacked, a player should stand up

Yes, but they always do for me when not using this PR. Has this already, somehow, been merged into the master branch or are there specific circumstances I need to work with?

**Comment by Kitzunu** [comment]
Ah no, it already works as expected, yes.

I just improved the logic to just check for players 🗡️ 

So if it works the same before the PR and after the PR then it is good =)

**Comment by tkn963** [comment]
> 
> 
> Ah no, it already works as expected, yes.
> 
> I just improved the logic to just check for players 🗡️
> 
> So if it works the same before the PR and after the PR then it is good =)

Oh, alright. My mistake. It's currently compiling the PR so I'll make sure in a little bit then.

**Comment by tkn963** [comment]
Works like it should 👍

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -12718,38 +12718,41 @@ void Unit::SetInCombatWith(Unit* enemy, uint32 duration)
     SetInCombatState(false, enemy, duration);
 }
 
-void Unit::CombatStart(Unit* target, bool initialAggro)
+void Unit::CombatStart(Unit* vicitm, bool initialAggro)
```
Comment (Efymer) [inline src/server/game/Entities/Unit/Unit.cpp:12721]: Shouldn't be `victim` instead of `vicitm`?

**Review by Efymer — CHANGES_REQUESTED** [review]

File: src/server/game/Entities/Unit/Unit.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/game/Entities/Unit/Unit.cpp:12721]: Ah yes typo, thanks

**Comment by FrancescoBorzi** [comment]
@Kitzunu conflicts again

**Review by Efymer — APPROVED** [review]

**Pull request merged**